File: ruby/pos_neg.rb
def pos_neg(a,b,negative)
	if (a<0 && b>0 && negative==false) || (a>0 && b<0 && negative==false)
		return true
		elsif a<0 && b<0 && negative==true
		return true
	else
		return false						
	end	
end
puts pos_neg(1,-1,false) ? "True" : "False"
puts pos_neg(-2,3,false) ? "True" : "False"
puts pos_neg(-4,-5,true) ? "True" : "False"
puts pos_neg(1,1,false) ? "True" : "False"
puts pos_neg(3,-9,true) ? "True" : "False"

File: php/pos_neg.php
<?php
	function pos_neg($a,$b,$negative)
	{
	
		return (($a<0 && $b>0) && $negative==false || ($a>0 && $b<0) && $negative==false || ($a<0 && $b<0) && $negative==true);
	}
	echo pos_neg(1,-1,false) ? "pass" : "fail"; echo "\n";
	echo pos_neg(-2,3,false) ? "pass" : "fail"; echo "\n";
	echo pos_neg(-4,-5,true) ? "pass" : "fail"; echo "\n";
	echo pos_neg(1,1,false) ? "fail" : "pass"; echo "\n";
	echo pos_neg(3,-9,true) ? "fail" : "pass"; echo "\n"
?>